app/views/shared/views/ems_common/_form.html.haml

```diff
@@ -28,7 +28,7 @@
         - else
           = h(@edit[:ems_types][@edit[:new][:emstype]])
     = render(:partial => "form_fields", :locals => {:url => url})
-    - if @edit[:new][:emstype] && @edit[:new][:emstype] == 'rhevm'
+    - if @edit[:new][:emstype] && %w(rhevm openstack_infra).include?(@edit[:new][:emstype])
```

Comment:
@chessbyte hm calling there @ems.supports_port? doesn't work for new provider. I guess the class of created provider is wrong here. Not even condition @ems.kind_of?(ManageIQ::Providers::Vmware::InfraManager) that is in that file works for new.

So that explains why is there only usage of type. :-)